xlb/operator/stepper/nse_stepper.py

```diff
@@ -102,68 +107,87 @@ class BoundaryConditionIDStruct:
             id_ZouHeBC_pressure: wp.uint8
             id_RegularizedBC_velocity: wp.uint8
             id_RegularizedBC_pressure: wp.uint8
+            id_ExtrapolationOutflowBC: wp.uint8
 
         @wp.func
         def apply_post_streaming_bc(
             f_pre: Any,
             f_post: Any,
+            f_aux: Any,
             missing_mask: Any,
             _boundary_id: Any,
             bc_struct: Any,
         ):
             # Apply post-streaming type boundary conditions
             if _boundary_id == bc_struct.id_EquilibriumBC:
                 # Equilibrium boundary condition
-                f_post = self.EquilibriumBC.warp_functional(f_pre, f_post, missing_mask)
```

Comment:
I think it's better that for BCs we drop warp_functional, and either use warp_poststream or post_collision. I think you can unify the way the BCs are applied if you only keep  warp_functional_poststream and post_collision., we the default returning f_post so they won't do anything.